src/eth_rpc/servers/debug_rpc.rs

```diff
@@ -47,7 +45,9 @@ impl<P: EthereumProvider + Send + Sync + 'static> DebugApiServer for DebugRpc<P>
         };
         let mut raw_block = Vec::new();
         if let Some(block) = block {
-            let block = rpc_to_primitive_block(block.inner).map_err(EthApiError::from)?;
+            let block = Block::try_from(block.inner)
+                .map_err(|_| EthereumDataFormatError::PrimitiveError)
+                .map_err(EthApiError::from)?;
```

Comment:
I think we can do both steps into a single `map_err` no?

Reply:
merged!